common/util/FileUtil.cpp

```diff
@@ -84,11 +105,17 @@ std::string get_file_path(const std::vector<std::string>& input) {
 }
 
 bool create_dir_if_needed(const std::string& path) {
-  if (!std::filesystem::is_directory(path)) {
-    std::filesystem::create_directories(path);
-    return true;
+  if (std::filesystem::is_directory(path)) {
+    return std::filesystem::create_directories(path);
+  } else {
+    // check if the "file" has an extension
+    auto temp = std::filesystem::path(path);
+    if (temp.has_extension()) {

```

Comment:
will this work if you have a folder with a `.` in this name? should we have a separate function for "make folders so I can save this file?"

Reply:
hmmm, ill have to check, maybe potentially reverting to a separate function for now yeah.  I just wanted to get away from the pattern of:
1. make directory with a path omitting the file name
2. then append the filename to the path and work with it

and instead just allow the function to intelligently make the directory properly even if it terminates with a file name.  But as you highlight....the edge-cases for this can be annoying / potentially untenable.